cfme/v2v/migration_plans.py

```diff
@@ -0,0 +1,636 @@
+        vms = self.migration_request_details_list.read()
+        for vm in vms:
+            clock_reading1 = self.migration_request_details_list.get_clock(vm)
+            time.sleep(1)  # wait 1 sec to see if clock is ticking
+            logger.info(
+                "For vm %s, current message is %s",
+                vm,
+                self.migration_request_details_list.get_message_text(vm),
+            )
+            logger.info(
+                "For vm %s, current progress description is %s",
+                vm,
+                self.migration_request_details_list.get_progress_description(vm),
+            )
+            clock_reading2 = self.migration_request_details_list.get_clock(vm)
+            logger.info("clock_reading1: %s, clock_reading2:%s", clock_reading1, clock_reading2)
+            logger.info(
+                "For vm %s, is currently in progress: %s",
+                vm,
+                self.migration_request_details_list.is_in_progress(vm),
+            )
+            migration_plan_in_progress_tracker.append(
+                self.migration_request_details_list.is_in_progress(vm)
+                and (clock_reading1 < clock_reading2)
+            )
+        return not any(migration_plan_in_progress_tracker)
+
+
+@attr.s
+class MigrationPlan(BaseEntity):
+    """Class representing v2v Migration Plan"""
+
+    name = attr.ib()
+    infra_map = attr.ib()
+    vm_list = attr.ib()
+    description = attr.ib(default=None)
+    csv_import = attr.ib(default=False)
+    target_provider = attr.ib(default=None)
+    osp_security_group = attr.ib(default=None)
+    osp_flavor = attr.ib(default=None)
+    pre_playbook = attr.ib(default=None)
+    post_playbook = attr.ib(default=None)
+
+    def is_plan_in_progress(self):
+        """MIQ V2V UI is going through redesign as OSP will be integrated."""
+        view = navigate_to(self, "InProgress")
+
+        def _in_progress():
+            try:
+                is_plan_visible = view.progress_card.is_plan_visible(self.name)
+                if is_plan_visible:
+                    try:
+                        plan_time_elapsed = view.progress_card.get_clock(self.name)
+                        new_msg = "time elapsed for migration: {time}".format(
+                            time=plan_time_elapsed
+                        )
+                    except NoSuchElementException:
+                        new_msg = "playbook is executing.."
+                        pass
+                    logger.info(
+                        "For plan {plan_name}, is plan in progress: {visibility}, {message}".format(
+                            plan_name=self.name, visibility=is_plan_visible, message=new_msg
+                        )
+                    )
+                return not is_plan_visible
+            except ItemNotFound:
+                return True
+
+        return wait_for(
+            func=_in_progress,
+            message="migration plan is in progress, be patient please",
```

Comment:
This message is a bit confusing, as well as the intent of this wait_for.  If the plan was visible, false is returned, and it waits. If the plan was not visible (exception, or just `is_plan_visible=False`), this stops waiting - the plan is then not in progress, but some other state?

This is a property or method for `is_in_progress`, why is it waiting for the plan to be not in visible?  This method appears to be doing much more than just returning whether or not the plan is in progress.

Reply:
Added docblock